_posts/2023-08-30-spiffe.md

```diff
@@ -0,0 +1,364 @@
+
+
+### Create a namespace for cert-manager
+
+We begin by creating a namespace where we will deploy cert-manager and SPIFFE.<br>
+Use the following two commands to create the namespaces and switch to the desired namespace:
+
+```bash
+kubectl create namespace cert-manager
+kubectl config set-context --current --namespace=cert-manager
+```
+
+### Add jetstack
+
+To install Cert-manager and SPIFFE, we use Helm. In order to install these Helm charts, we need to fetch the Jetstack repository locally.<br>
+Use the following commands to achieve this:
+
+```bash
+helm repo add jetstack https://charts.jetstack.io
+helm repo update
+```
+
+### Install cert manager
+
+Now that we can install packages from Jetstack, we can proceed to install Cert Manager on our Kubernetes cluster.<br>
+Use the following commands to install Cert Manager:
+
+```bash
+kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.11.0/cert-manager.crds.yaml
+helm upgrade --install cert-manager jetstack/cert-manager --version v1.11.0 --set prometheus.enabled=false
+```
+
+
+## Install SPIFFE
+
+### Install SPIFFE
+
+Now that Cert Manager is installed, we can proceed to the installation of SPIFFE.<br>
+To do this, use the following command:
+
+Please note: for the trusted domain, you could use a different name; I suggest using the name of your Kubernetes cluster. You will need this domain name later when setting up AWS policies.
+
+```bash
+helm upgrade --install cert-manager-csi-driver-spiffe jetstack/cert-manager-csi-driver-spiffe --wait \
+--set image.tag=aws \
+--set image.repository.driver=ghcr.io/joshvanl/cert-manager-csi-driver \
+--set image.repository.approver=ghcr.io/joshvanl/cert-manager-csi-driver-approver \
+--set "app.logLevel=1" \
+--set "app.trustDomain=demo.home.cluster" \
+--set "app.approver.signerName=clusterissuers.cert-manager.io/csi-driver-spiffe-ca" \
+--set "app.issuer.name=csi-driver-spiffe-ca" \
+--set "app.issuer.kind=ClusterIssuer" \
+--set "app.issuer.group=cert-manager.io"
+```
+
+
+### Configure SPIFFE
+
+To obtain and renew certificates automatically, we use the following ClusterIssuer.<br>
+Deploy it using the following command:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/nicholasM95/spiffe-demo/main/setup/clusterissuer.yaml
+```
+
+### Verify installation
+
+Verify if all pods have started and check whether the secret csi-driver-spiffe-ca has been created.
+
+<img src="{{ '/img/2023-08-30-spiffe/spiffe-1.png' | prepend: site.baseurl }}" alt="Verify spiffe installation" class="image fit" style="margin:0px auto; max-width:100%">
+
+## Configure AWS
+### Create the s3 bucket
+
+Create an S3 bucket with the name 'demospiffe' and upload a text file named 'hello.txt'. Our application will read this file later.
```

Comment:
```suggestion
Create a S3 bucket with the name 'demospiffe' and upload a text file named 'hello.txt'. Our application will read this file later.
```

Reply:
Bucket names are global and not restricted to an account, so it might give off the wrong idea if it says to create a bucket with a specific name rather than letting the user decide one.